analysis_runner/dataproc.py

```diff
@@ -21,6 +22,102 @@
 PYFILES_ZIP = 'pyfiles.zip'
 
 
+class DataprocCluster:
+    """
+    Helper class that represents a Dataproc cluster created within a Batch
+    """
+
+    def __init__(
+        self,
+        batch: hb.Batch,
+        start_job: hb.batch.job.Job,
+        stop_job: hb.batch.job.Job,
+        cluster_name: str,
+    ):
+        self.batch = batch
+        self.start_job = start_job
+        self.stop_job = stop_job
+        self.cluster_name = cluster_name
+
+    def add_job(
+        self,
+        script: str,
+        job_name: Optional[str] = None,
+        pyfiles: Optional[List[str]] = None,
+    ):
+        """
+        Create a job that submits the `script` to the cluster
+        """
+        job = _add_submit_job(
+            batch=self.batch,
+            cluster_name=self.cluster_name,
+            script=script,
+            job_name=job_name,
+            pyfiles=pyfiles,
+        )
+        job.depends_on(self.start_job)
+        self.stop_job.depends_on(job)
+
+
+def hail_dataproc(
+    batch: hb.Batch,
+    *,
+    max_age: str,
+    num_workers: int = 2,
+    num_secondary_workers: int = 0,
+    worker_machine_type: Optional[str] = None,  # e.g. 'n1-highmem-8'
+    worker_boot_disk_size: Optional[int] = None,  # in GB
+    secondary_worker_boot_disk_size: Optional[int] = None,  # in GB
+    packages: Optional[List[str]] = None,
+    init: Optional[List[str]] = None,
+    vep: Optional[str] = None,
+    requester_pays_allow_all: bool = True,
+    depends_on: Optional[List[hb.batch.job.Job]] = None,
+    job_name: Optional[str] = None,
+    scopes: Optional[List[str]] = None,
+    labels: Optional[Dict[str, str]] = None,
+) -> DataprocCluster:
+    """
+    Adds jobs to the batch that start and stop a Dataproc cluster, and returns
+    a DataprocCluster object with a submit() method that allows to add jobs into
+    this cluster that submit Query scripts.
```

Comment:
```suggestion
    a DataprocCluster object with an add_job() method to submit Hail Query scripts.
```

Reply:
I wanted to be explicit that we don't actually submit Query scripts here. We add a job will submit.

Reply:
But fixed the method name!